OSDOCS-12055 OCP 4.15.34 Release Notes (9/25 release) (#82291)
openshift/openshift-docs: release_notes/ocp-4-15-release-notes.adoc

----

[id="ocp-4-15-34-enhancements_{context}"]
==== Enhancements

The following enhancement is included in this z-stream release:

[id="ocp-4-15-34-IO-new-metric_{context}"]
===== Insights Operator collects new metric

* The Insights Operator (IO) can now collect data from the `haproxy_exporter_server_threshold` metric.
(link:https://issues.redhat.com/browse/OCPBUGS-38593[*OCPBUGS-38593*])

[id="ocp-4-15-34-bug-fixes_{context}"]
==== Bug fixes

* Previously, Operator Lifecycle Manager (OLM) catalog source pods did not recover from node failure if the `registryPoll` field was `none`. With this release, OLM `CatalogSource` registry pods recover from cluster node failures and the issue is resolved. (link:https://issues.redhat.com/browse/OCPBUGS-41981[*OCPBUGS-41981*])

* Previously, if you created a hosted cluster by using a proxy for the purposes of making the cluster reach a control plane from a compute node, the compute node would be unavailable to the cluster. With this release, the proxy settings are updated for the node so that the node can use a proxy to successfully communicate with the control plane. (link:https://issues.redhat.com/browse/OCPBUGS-41947[*OCPBUGS-41947*])

> * Previously, When the Operator Lifecycle Manager (OLM) evaluated a potential upgrade, it used the dynamic client list for all custom resource (CR) instances in the cluster.  For clusters with a large number of CRs, that could result in timeouts from the apiserver and stranded upgrades. With this release, the issue is resolved. (link:https://issues.redhat.com/browse/OCPBUGS-41819[*OCPBUGS-41819*])

Reviewer: Update is the preferred term per our term glossary:

```suggestion
* Previously, when the Operator Lifecycle Manager (OLM) evaluated a potential update, it used the dynamic client list for all custom resource (CR) instances in the cluster.  For clusters with a large number of CRs, that could result in timeouts from the apiserver and stranded updates. With this release, the issue is resolved. (link:https://issues.redhat.com/browse/OCPBUGS-41819[*OCPBUGS-41819*])
```

> * Previously, an `AdditionalTrustedCA` field that was specified in the Hosted Cluster image configuration was not reconciled into the openshift-config namespace as expected and the component was not available. With this release, the issue is resolved. (link:https://issues.redhat.com/browse/OCPBUGS-41809[*OCPBUGS-41809*])

Reviewer: I feel like this value belongs in backticks, but if you're not sure then you can ignore this:
```suggestion
* Previously, an `AdditionalTrustedCA` field that was specified in the Hosted Cluster image configuration was not reconciled into the `openshift-config` namespace as expected and the component was not available. With this release, the issue is resolved. (link:https://issues.redhat.com/browse/OCPBUGS-41809[*OCPBUGS-41809*])
```

* Previously, when you created a `LoadBalancer` service for the Ingress Operator, a log message was generated that stated the change was not effective. This log message should only trigger for a change to an Infra custom resource. With this release, this log message is no longer generated when you create a `LoadBalancer` service for the Ingress Operator. (link:https://issues.redhat.com/browse/OCPBUGS-41635[*OCPBUGS-41635*])

> * Previously, if an IP was assigned to an egress node and was deleted, then pods selected by that egress IP might have had incorrect routing information to that egress node. With this release, the issue has been resolved. (link:https://issues.redhat.com/browse/OCPBUGS-41340[*OCPBUGS-41340*])

Reviewer: ```suggestion
* Previously, if an IP address was assigned to an egress node and was deleted, then pods selected by that egress IP address might have had incorrect routing information to that egress node. With this release, the issue has been resolved. (link:https://issues.redhat.com/browse/OCPBUGS-41340[*OCPBUGS-41340*])
```

> * Previously, proxying for Operators that run in the control plane of a hosted cluster was done through proxy settings on the Konnectivity agent pod that runs in the data plane. As a consequence, it was not possible to distinguish whether proxying was needed based on application protocol. For parity with OpenShift Container Platform, IDP communication via HTTPS or HTTP should be proxied, but LDAP communication should not be proxied. This type of proxying also ignores `NO_PROXY` entries that rely on host names because by the time traffic reaches the Konnectivity agent, only the destination IP address is available. With this release, in hosted clusters, proxy is invoked in the control plane via `konnectivity-https-proxy` and `konnectivity-socks5-proxy`, and proxying traffic is stopped from the Konnectivity agent. As a result, traffic that is destined for LDAP servers is no longer proxied. Other HTTPS or HTTPS traffic is proxied correctly. The `NO_PROXY` setting is honored when you specify hostnames. (link:https://issues.redhat.com/browse/OCPBUGS-38065[*OCPBUGS-38065*])

Reviewer: ```suggestion
* Previously, proxying for Operators that run in the control plane of a hosted cluster was done through proxy settings on the Konnectivity agent pod that runs in the data plane. As a consequence, it was not possible to distinguish whether proxying was needed based on application protocol. For parity with {product-title}, IDP communication via HTTPS or HTTP should be proxied, but LDAP communication should not be proxied. This type of proxying also ignores `NO_PROXY` entries that rely on host names because by the time traffic reaches the Konnectivity agent, only the destination IP address is available. With this release, in hosted clusters, proxy is invoked in the control plane via `konnectivity-https-proxy` and `konnectivity-socks5-proxy`, and proxying traffic is stopped from the Konnectivity agent. As a result, traffic that is destined for LDAP servers is no longer proxied. Other HTTPS or HTTPS traffic is proxied correctly. The `NO_PROXY` setting is honored when you specify hostnames. (link:https://issues.redhat.com/browse/OCPBUGS-38065[*OCPBUGS-38065*])
```

> * Previously, proxying for Operators that run in the control plane of a hosted cluster was done through proxy settings on the Konnectivity agent pod that runs in the data plane. As a consequence, it was not possible to distinguish whether proxying was needed based on application protocol. For parity with OpenShift Container Platform, IDP communication via HTTPS or HTTP should be proxied, but LDAP communication should not be proxied. This type of proxying also ignores `NO_PROXY` entries that rely on host names because by the time traffic reaches the Konnectivity agent, only the destination IP address is available. With this release, in hosted clusters, proxy is invoked in the control plane via `konnectivity-https-proxy` and `konnectivity-socks5-proxy`, and proxying traffic is stopped from the Konnectivity agent. As a result, traffic that is destined for LDAP servers is no longer proxied. Other HTTPS or HTTPS traffic is proxied correctly. The `NO_PROXY` setting is honored when you specify hostnames. (link:https://issues.redhat.com/browse/OCPBUGS-38065[*OCPBUGS-38065*])

Reviewer: Per [ISG word usage](https://www.ibm.com/docs/en/ibm-style?topic=word-usage#v), "via" should be avoided: 
```suggestion
* Previously, proxying for Operators that run in the control plane of a hosted cluster was done through proxy settings on the Konnectivity agent pod that runs in the data plane. As a consequence, it was not possible to distinguish whether proxying was needed based on application protocol. For parity with OpenShift Container Platform, IDP communication through HTTPS or HTTP should be proxied, but LDAP communication should not be proxied. This type of proxying also ignores `NO_PROXY` entries that rely on host names because by the time traffic reaches the Konnectivity agent, only the destination IP address is available. With this release, in hosted clusters, proxy is invoked in the control plane through `konnectivity-https-proxy` and `konnectivity-socks5-proxy`, and proxying traffic is stopped from the Konnectivity agent. As a result, traffic that is destined for LDAP servers is no longer proxied. Other HTTPS or HTTPS traffic is proxied correctly. The `NO_PROXY` setting is honored when you specify hostnames. (link:https://issues.redhat.com/browse/OCPBUGS-38065[*OCPBUGS-38065*])
```
